test/jdk/java/awt/color/ICC_Profile/BuiltInProfileCheck/BuiltInProfileCheck.java

```diff
@@ -0,0 +1,165 @@
+import java.io.ObjectOutputStream;
+import java.util.Map;
+
+public class BuiltInProfileCheck {
+    private static final int HEADER_TAG = ICC_Profile.icSigHead;
+    private static final int INDEX = ICC_Profile.icHdrDeviceClass;
+    private static final String EXCEPTION_MSG = "Built-in profile cannot be modified";
+    private static ICC_Profile testProfile;
+
+    private static final Map<Integer, String> colorSpace = Map.of(
+            ColorSpace.CS_sRGB, "CS_sRGB",
+            ColorSpace.CS_PYCC, "CS_PYCC",
+            ColorSpace.CS_GRAY, "CS_GRAY",
+            ColorSpace.CS_CIEXYZ, "CS_CIEXYZ",
+            ColorSpace.CS_LINEAR_RGB, "CS_LINEAR_RGB"
+    );
+
+    public static void main(String[] args) throws Exception {
+        System.out.println("CASE 1: Testing BuiltIn Profile");
+        for (int cs : colorSpace.keySet()) {
+            prepareTestProfile("BuiltIn", true, cs);
+            testProfile(true, cs);
+        }
+        System.out.println("Passed\n");
+
+        System.out.println("CASE 2: Testing Custom Profile");
+        prepareTestProfile("Custom", false, ColorSpace.CS_sRGB);
+        testProfile(false, ColorSpace.CS_sRGB);
+        System.out.println("Passed\n");
+
+        System.out.println("CASE 3: Testing Built-In Profile"
+                            + " Serialization & Deserialization");
+        for (int cs : colorSpace.keySet()) {
+            prepareTestProfile("SerializeBuiltIn", true, cs);
+            testProfile(true, cs);
+        }
+        System.out.println("Passed\n");
+
+        System.out.println("CASE 4: Testing Custom Profile"
+                            + " Serialization & Deserialization");
+        prepareTestProfile("SerializeCustom", false, ColorSpace.CS_sRGB);
+        testProfile(false, ColorSpace.CS_sRGB);
+        System.out.println("Passed\n");
+
+        System.out.println("CASE 5: Test reading Built-In profile from .icc file");
+        prepareTestProfile("ReadBuiltIn", true, ColorSpace.CS_sRGB);
+        testProfile(true, ColorSpace.CS_sRGB);
+        System.out.println("Passed\n");
+
+        System.out.println("CASE 6: Test reading Custom profile from .icc file");
+        prepareTestProfile("ReadCustom", false, ColorSpace.CS_sRGB);
+        testProfile(false, ColorSpace.CS_sRGB);
+        System.out.println("Passed\n");
+    }
+
+    private static void prepareTestProfile(String testCase,
+                                           boolean isBuiltIn, int cs) {
+        ICC_Profile builtInProfile = ICC_Profile.getInstance(cs);
+        // if isBuiltIn=true use builtInProfile else create a copy
+        testProfile = isBuiltIn
+                      ? builtInProfile
+                      : ICC_Profile.getInstance(builtInProfile.getData());
+
+        switch (testCase) {
+            case "BuiltIn", "Custom" -> {
+                // empty case block
+                // no further processing of testProfile required in these 2 cases.
+            }
```

Comment:
I could move this to a if block before switch and simply return but since switch-case is used I thought to keep all the cases at one place.